subscription/src/main/java/org/killbill/billing/subscription/api/user/DefaultSubscriptionBaseApiService.java

```diff
@@ -512,6 +515,14 @@ private void doChangePlan(final DefaultSubscriptionBase subscription,
         if (nextPhaseEvent != null) {
             events.add(nextPhaseEvent);
         }
+        if(nextTimedPhase == null && currentPhase.getPhase().getPhaseType() == PhaseType.FIXEDTERM) {
```

Comment:
Suggestion: Instead of `if(nextTimedPhase == null && currentPhase.getPhase().getPhaseType() == PhaseType.FIXEDTERM) {` you could just leverage the existing `if` and have a `else if ( currentPhase.getPhase().getPhaseType() == PhaseType.FIXEDTERM)`.

Nit: Also formatting-wise we typically have a space between `if` and `(`

Reply:
Fixed